From 583021306b34b9860e61b60d4e026b563a04a019 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Mon, 24 Feb 2014 20:59:47 +0100
Subject: [PATCH 30/30] BUG/MINOR: config: fix a crash on startup when a
 disabled backend references a peer

Disabled backends don't have their symbols resolved. We must not initialize
their peers section since they're not valid and instead still contain the
section's name.

There are other places where such unions are still in use, and other similar
errors might still happen. Ideally we should get rid of all of them in the
quite sensible config stage.
---
 src/cfgparse.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/cfgparse.c b/src/cfgparse.c
index 88e86d8..f13b967 100644
--- a/src/cfgparse.c
+++ b/src/cfgparse.c
@@ -7815,6 +7815,9 @@ out_uri_auth_compat:
 	 * other proxies.
 	 */
 	for (curproxy = proxy; curproxy; curproxy = curproxy->next) {
+		if (curproxy->state == PR_STSTOPPED)
+			continue;
+
 		if (!stktable_init(&curproxy->table)) {
 			Alert("Proxy '%s': failed to initialize stick-table.\n", curproxy->id);
 			cfgerr++;
-- 
1.8.3.2

